### PR TITLE
[YAML] Add a prompt command to the LogCommands cluster

### LINKED ISSUE #11853: [Feature] Add UserPrompt command to YAML tests.
#### Problem
Scripters need a way to add user prompts to their YAML tests.

Currently there is only a log entry available in YAML like:
```
tests:
    - label: "Log a simple message"
      cluster: "LogCommands"
      command: "Log"
      arguments:
          values:
              - name: "message"
                value: "This is a simple message"
```

 #### Proposed Solution
Similarly to the `command: Log`, please add a `command: UserPrompt` or something to that equivalent so scripters can distinguish between lines they are trying to log and prompts they are trying to display to the user.

Please follow something like: 

```
tests:
    - label: "Display a user prompt."
      cluster: "LogCommands"
      command: "Prompt"
      arguments:
          values:
              - name: "message"
                value: "Please issue a ____ cluster from the DUT controller."
```

BUT append a prefix to the value like `"USER_PROMPT:"` so that log lines which are intended to be user prompts are clearly defined. 

So the logged output from that above example should look like: 

`USER_PROMPT: Please issue a ____ cluster from the DUT controller`

### PR DESCRIPTION
**Pull request by vivien-apple**
#### Problem

Needs a log command with a prefix that is always the same.

#### Change overview
* Add a `prompt` command to `LogCommands`

#### Testing
It has been added to `TestLogCommands.yaml`

#fixes #11853

### CODE REVIEW

File: src/app/zap-templates/common/simulated-clusters/TestLogCommands.js [hunk 1]
```diff
@@ -30,9 +30,15 @@ const Log = {
   response : { arguments : [] }
 };
 
+const Prompt = {
```
Comment (trombadore) [inline src/app/zap-templates/common/simulated-clusters/TestLogCommands.js:33]: Can you actually change the this label from Prompt to `UserPrompt` instead, like: 

``` 
const UserPrompt = {
  name : 'UserPrompt',
  arguments : [ { type : 'CHAR_STRING', name : 'message' } ],
  response : { arguments : [] }
};
```

**Review by trombadore-google — CHANGES_REQUESTED** [review]

**Review by bzbarsky-apple — APPROVED** [review]

File: examples/chip-tool/commands/tests/TestCommand.cpp [hunk 2]
```diff
@@ -64,6 +64,13 @@ CHIP_ERROR TestCommand::Log(const char * message)
     return CHIP_NO_ERROR;
 }
 
+CHIP_ERROR TestCommand::UserPrompt(const char * message)
+{
+    ChipLogDetail(chipTool, "USER_PROMPT: %s", message);
+    WaitForMs(0);
```
Comment (andy31415) [inline examples/chip-tool/commands/tests/TestCommand.cpp:70]: why do we have this wait? is this for some flush? please add a comment or remove - waiting 0 seems to not make much sense.

**Review by andy31415 — APPROVED** [review]

**Review by trombadore-google — APPROVED** [review]

File: examples/chip-tool/commands/tests/TestCommand.cpp (hunk 2 above)
Comment (trombadore) [reply examples/chip-tool/commands/tests/TestCommand.cpp:70]: +1

Comment (vivien-apple) [reply examples/chip-tool/commands/tests/TestCommand.cpp]: I did a method named `ContinueOnChipMainThread` which is why there are those `WaitForMs(0)` here. We basically want the `NextTest` to be called onto the chip thread

**Review by woody-apple — APPROVED** [review]

**Comment by woody-apple** [comment]
Fast tracking given this enables tests.

**Pull request merged**